components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -161,15 +158,85 @@ class Nimbus : NimbusApi {
 
     override fun updateExperiments() {
         if (!isInitialized) return
-        nimbus.updateExperiments()
-        onExperimentUpdated?.invoke(nimbus.getActiveExperiments())
+        // We need to launch this off of the main thread because it involves a network request
+        // within the nimbus-sdk.
+        scope.launch {
```

Comment:
nit: I saw that nimbus creates its own single thread dispatcher. I wonder if that is wasteful, having a thread hang around just for nimbus. Using the IO dispatcher is not an option?

Reply:
I think we were more interested in the single-threadedness of the dispatcher for keeping async issues in check, but perhaps there's a lighter weight way of accomplishing this?